### ISSUE TITLE
[courier] courier-mta missing systemd init file

### ISSUE THREAD

**Issue by markc** [oerdnj/deb.sury.org#202]
courier-mta also needs a systemd init file otherwise the default incoming port 25 is not available.
```
~ cat /lib/systemd/system/courier-mta.service
[Unit]
Description=Courier ESMTP Daemon
Requires=courier-authdaemon.service
After=courier-authdaemon.service

[Service]
Type=forking
ExecStart=/usr/sbin/esmtpd start
ExecStop=/usr/sbin/esmtpd stop
PIDFile=/var/run/courier/esmtpd.pid

[Install]
WantedBy=multi-user.target
```

**Comment by oerdnj** [comment]
Do you have an idea whether it's possible to tell the courier daemon to not fork on start?

**Comment by markc** [comment]
No idea really but there is no "courier" daemon. Perhaps you mean the esmptd daemon?

**Comment by oerdnj** [comment]
I meant "courier daemon" as any courier daemon there is (authdaemon, imapd, pop3d, esmtpd, and others)

**Comment by markc** [comment]
Oh, if you said "courier daemons" I probably would have known what you meant. No, I don't know how to prevent any of them forking on startup. strace -f will follow the forked processes anyway.

If you followed the other issue you would see that submit does a setgid(1) and setuid(1) (daemon:daemon) before trying to cd into /var/lib/courier/tmp which is owned by courier. I'm pretty sure that is where the problem lies. The 1:1 (set)uid:gid is compiled in whereas the courier uid:gid is dynamically allocated when the packages are installed. I've tried adding courier to the daemon group, and vice versa, and chmod 775 everything in /var/lib/courier but still no joy.

**Comment by oerdnj** [comment]
I already have almost working patch, but the way autotools are used[1] in this package (and the fact I am regenerating most of the files) makes the build to take a very long time, that makes debugging the packages a long process.

1. really - putting configure.ac to every sub-directory is just an abomination and not a build system

**Comment by oerdnj** [comment]
Included in `courier - 0.75.0-7+deb.sury.org~<dist>+2`.
